app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,85 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(_type, _id, data)
+      # Temporarily only supporting atomic.
+      raise 'Service Type composite not supported' if data['service_type'] == 'composite'
+      create_atomic(data).tap(&:save!)
+    rescue => err
+      raise BadRequestError, "Could not create Service Template - #{err}"
+    end
+
+    private
+
+    def create_atomic(data)
+      service_template = ServiceTemplate.new(data.except('request_info'))
+      request_info = data['request_info']
+      if service_template.prov_type.include?('generic')
+        # Configure generic service template
+        add_orchestration_template_vars(service_template, request_info)
+        add_ansible_tower_job_template_vars(service_template, request_info)
+      else
+        # Configure non-generic service template
+        service_template.add_resource(create_service_template_request(request_info))
+      end
+      dialog = Dialog.find(request_info['dialog_id']) if request_info && request_info['dialog_id']
+      set_provision_action(service_template, dialog, request_info)
+      set_retirement_reconfigure_action(service_template, dialog, request_info)
+      service_template
+    end
+
+    # Need to set the request for non-generic Service Template
+    def create_service_template_request(request_data)
+      # data must be symbolized
+      request_params = request_data.symbolize_keys
+      wf = MiqProvisionWorkflow.class_for_source(request_params[:src_vm_id]).new(request_params, @auth_user_obj)
+      raise 'Could not find Provision Workflow class for source VM' unless wf
+      request = wf.make_request(nil, request_params)
+      raise 'Could not create valid request' if request == false || !request.valid?
+      request
+    end
+
+    def set_provision_action(service_template, dialog, request_info)
```

Comment:
Mostly separated because the provisioning entrypoint takes in the service type, the others don't, and when I put them together rubocop yelled about complexity. Will try and refactor though.